components/brave_wallet/browser/eth_tx_state_manager.cc

```diff
@@ -109,11 +114,14 @@ std::unique_ptr<TxMeta> EthTxStateManager::ValueToTxMeta(
   return meta;
 }
 
-std::string EthTxStateManager::GetTxPrefPathPrefix() {
-  return base::StrCat(
-      {kEthereumPrefKey, ".",
-       GetNetworkId(prefs_, mojom::CoinType::ETH,
-                    json_rpc_service_->GetChainId(mojom::CoinType::ETH))});
+std::string EthTxStateManager::GetTxPrefPathPrefix(
+    const absl::optional<std::string>& chain_id) {
+  if (chain_id.has_value()) {
+    return base::StrCat(
+        {kEthereumPrefKey, ".",
+         GetNetworkId(prefs_, mojom::CoinType::ETH, *chain_id)});
+  }
+  return kEthereumPrefKey;
```

Comment:
Is it possible that chain_id is a nullopt here? If yes, what kind transactions is supposed to be stored under root kEthereumPrefKey ?

Reply:
There won't be any transactions stored under coin key and never will be. It is only used in `TxStateManager::WipeTxs()` to clear all transactions across all chain ids under this coin key.

Reply:
We can remove `WipeTxs` though since it is not used in production code. So it might ease confusion with a mandatory `chain_id` input.

Reply:
It confused me too, I'm fine if we have comments above the declaration of `TxStateManager::GetTxPrefPathPrefix` or remove `WipeTxs` and make it required if we don't need `WipeTxs`.
Personally I prefer just remove `WipeTxs` and make chain_id non-optional since that makes the code a little bit simpler here.

Reply:
Actually, it seems `GetTransactionsByStatus` might be calling this with nullopt as chain_id too.

Reply:
It confused me too, I'm fine if we have comments above the declaration of `TxStateManager::GetTxPrefPathPrefix` or remove `WipeTxs` and make it required if we don't need `WipeTxs`.
Personally I prefer just remove `WipeTxs` and make chain_id non-optional since that makes the code a little bit simpler here.

Reply:
Actually, it seems `GetTransactionsByStatus` might be calling this with nullopt as chain_id too so we might need to keep it optional here.

Reply:
Ah, right I forgot that use case, I will add comment to that method saying it is all chain id under the coin key.

Reply:
fixed in fedd8a6e93192ead54bde1ad8a99896f9b15f5f7

Reply:
LGTM, but will defer for @supermassive to resolve this.